tasks/Task.cpp

```diff
@@ -13,6 +14,55 @@ Task::~Task()
 {
 }
 
+PWMTable loadCommandToPWMTable(std::string const& csv_file_path)
+{
+
+    PWMTable table;
+
+    std::ifstream csv_file(csv_file_path);
+
+    if (!csv_file.is_open()) {
+        throw std::runtime_error("could not open csv file");
+    }
+
+    std::string row;
+    while (std::getline(csv_file, row)) {
+        std::size_t comma_i = row.find(',');
+        table.cmd.push_back(std::atof(row.substr(0, comma_i).c_str()));
+        table.duty_cycle_width.push_back(std::atoi(row.substr(comma_i + 1).c_str()));
+    }
+
+    return table;
+}
+
+uint32_t Task::computePWMCommand(float command) const
+{
+    if (command > m_lower_deadzone_command && command < m_upper_deadzone_command) {
+        return m_no_actuation_pwm_command;
+    }
```

Comment:
The purpose of this component is to linearize the command space. There should be no deadzone on the input space, only on the output space. Dead zones are good for analog systems. We don't need them here where '0' is *exactly* zero.

What you should have is

```
if (command == 0) {
    return m_no_actuation_pwm_command;
}
```

To make sure you pick up the "best" zero PWM.

Reply:
Just to be sure: it is guaranteed that the command is 0, or out of the dead zone otherwise?

Reply:
sorry to insist, I really don't get it

> If cmd != 0, the interpolation will find the right place.

Is it ok if the found pwm lies inside the deadzone, that is 1480 < pwm < 1520 ?

Reply:
* If cmd != 0, the found PWM is not within the dead zone by *definition*.
* If cmd == 0, the found PWM should be m_no_actuation_pwm_command, which should really be the center of the dead zone, that is the zone where `effort == 0` (and not a property)